### ISSUE TITLE
[Feature request] Please add a button to switch keyboards

### ISSUE THREAD

**Issue by asereze** [AnySoftKeyboard/AnySoftKeyboard#2386]
Since AnySoftKeyboard doesn't have all the languages that are available in the AOSP keyboard, or in some other ones, it would be great to have in the keyboard a button like the globe one that it's available on AOSP or in the Google Keyboard, to switch to another keyboard. To make an example, I write in Sardinian, Italian, English, Portuguese and Japanese. For the first three I can use AnySoftKeyboard, but for Japanese I need to go to the system's settings every time and deactivate it to be able to use the Google Keyboard (that language is missing on AOSP as well). That's quite annoying, and in my opinion it could be a problem for AnySoftKeyboard, because it could motivate someone to exclusively use the keyboard that has most or all of the languages available, to avoid having to do that all the time.

**Comment by azhao12345** [comment]
I was looking at the api to make this change this earlier.  Not familiar with the ASK logic though, so not sure where to start with adding that call.
https://developer.android.com/guide/topics/text/creating-input-method#Switching

**Comment by nedy13** [comment]
I think you can use Android function and there is no additional option needed inside the keyboard. I think there are two different ways available. You have a option in the notification bar or you have an icon to switch input method in the bottom action bar.

![Screenshot_20201105-180144_Firefox~2](https://user-images.githubusercontent.com/4283170/98273818-f9b4e400-1f92-11eb-8886-963f6b7c87de.png)

Then you can choose your keyboard.

![Screenshot_20201105-181131_Firefox~2](https://user-images.githubusercontent.com/4283170/98273986-2f59cd00-1f93-11eb-8daa-0fcaf7f41257.png)

Hope this helps.

**Comment by O-DoC** [comment]
> I think you can use Android function and there is no additional option needed inside the keyboard. I think there are two different ways available. You have a option in the notification bar or you have an icon to switch input method in the bottom action bar.
> 
> ![Screenshot_20201105-180144_Firefox~2](https://user-images.githubusercontent.com/4283170/98273818-f9b4e400-1f92-11eb-8886-963f6b7c87de.png)
> 
> Then you can choose your keyboard.
> 
> ![Screenshot_20201105-181131_Firefox~2](https://user-images.githubusercontent.com/4283170/98273986-2f59cd00-1f93-11eb-8daa-0fcaf7f41257.png)
> 
> Hope this helps.

I used the same button all the time (switch to keepass keyboard) !
No need new function as Android already handle this natively 👍🏽

**Comment by asereze** [comment]
@nedy13 Ok, thank you very much! I think that button is only available on some Android versions (I use Android 9 - LineageOS 16.0 and I don't have it), but it works from the notification, though!

**Comment by O-DoC** [comment]
@asereze 
You could check something in `Settings -> Accessibility -> Interaction Controls -> Accesibility -> Use Service` if it's `On` it could break multi-keyboard selection bouton. 
Hope it's help.

**Comment by Jonta** [comment]
@asereze Did @O-DoC's suggestion work for you? =)

**Comment by asereze** [comment]
@Jonta No, I don't have that option on my phone, or at least not in that menu. But I'm using the option in the notification bar, like @nedy13 suggested me to, and it works perfectly.

**Comment by c933103** [comment]
> @asereze 
> You could check something in `Settings -> Accessibility -> Interaction Controls -> Accesibility -> Use Service` if it's `On` it could break multi-keyboard selection bouton. 
> Hope it's help.

But the button would go away when user need to use accessibility service enabled those accessibility services according to personal needs

**Comment by Zocker1999NET** [comment]
Also devices with hardware buttons cannot access the native change button as there is none. However you should be able to enable a persistent notification if a keyboard is displayed and use that to change the keyboard (LineageOS 18.1, Android 11). And I also can access the menu using ASK by opening the utility keyboard using a gesture. However I would still like to have either a button or support opening the menu directly using a gesture for faster access.

**Comment by nicoursi** [comment]
You could also long press on `ENTER` and then tap on "change input"

**Comment by JimmyCushnie** [comment]
> You could also long press on `ENTER` and then tap on "change input method"

Thank you!!

The convention is to expose this functionality on a long press of Spacebar. OpenBoard, for example, does it this way. I wonder if that same change could be brought to AnySoft?
